service/src/main/java/datawave/microservice/AccumuloConnectionService.java

```diff
@@ -0,0 +1,220 @@
+    }
+    
+    /**
+     * Returns a Connection representing the connection to accumulo configured with the specified parameters
+     *
+     * @param metadataTable
+     *            the name of the metadata table to use in the connection
+     * @param modelTable
+     *            the name of the model table to use in the connection
+     * @param modelName
+     *            the name of the model to use in the connection
+     * @param user
+     *            the user (and the authorities and other security aspects related to the user)
+     *            
+     * @return a Connection representing the connection to accumulo
+     */
+    public Connection getConnection(String metadataTable, String modelTable, String modelName, DatawaveUserDetails user) {
+        Connection connection = new Connection();
+        connection.setMetadataTable(getSupplierValueIfBlank(metadataTable, dataDictionaryConfiguration::getMetadataTableName));
+        connection.setModelTable(getSupplierValueIfBlank(modelTable, dataDictionaryConfiguration::getModelTableName));
+        connection.setModelName(getSupplierValueIfBlank(modelName, dataDictionaryConfiguration::getModelName));
+        connection.setAuths(getAuths(user));
+        connection.setAccumuloClient(accumuloClient);
+        return connection;
+    }
+    
+    private String getSupplierValueIfBlank(final String value, final Supplier<String> supplier) {
+        return StringUtils.isBlank(value) ? supplier.get() : value;
+    }
+    
+    /**
+     * Return the accumulo authorizations for the user
+     *
+     * @param currentUser
+     *            the user for whom to get the authorizations
+     *            
+     * @return the accumulo authorizations for the user
+     */
+    public Set<Authorizations> getAuths(DatawaveUserDetails currentUser) {
+        //@formatter:off
+        return currentUser.getProxiedUsers().stream()
+                .map(DatawaveUser::getAuths)
+                .map(a -> new Authorizations(a.toArray(new String[0])))
+                .collect(Collectors.toSet());
+        //@formatter:on
+    }
+    
+    /**
+     * Return the authorizations for the specified user downgraded to the specified level
+     *
+     * @param requestedAuthorizations
+     *            the level of authorizations to downgrade to
+     * @param currentUser
+     *            the user for whom to downgrade the authorizations
+     *            
+     * @return the authorizations for the specified user downgraded to the specified level
+     */
+    public Set<Authorizations> getDowngradedAuthorizations(String requestedAuthorizations, DatawaveUserDetails currentUser) {
+        DatawaveUser primaryUser = currentUser.getPrimaryUser();
+        return userAuthFunctions.mergeAuthorizations(userAuthFunctions.getRequestedAuthorizations(requestedAuthorizations, currentUser.getPrimaryUser()),
```

Comment:
second call to currentUser.getPrimaryUser() could use primaryUser variable.

Reply:
agreed.  fixing in next commit.